### PR TITLE
Fix(domain-record): TXT record and helpers

### PR DESCRIPTION
**Pull request by Monitob**
* Fix schema domain record
* Fix read method
* Fix helpers on TXT record DATA
* Fix setter ID for resource domain_record
* Fix add missing attributes to update process

### CODE REVIEW

File: scaleway/resource_domain_record_test.go [hunk 1]
```diff
@@ -308,39 +386,6 @@ func TestAccScalewayDomainRecord_HTTPService(t *testing.T) {
 		ProviderFactories: tt.ProviderFactories,
 		CheckDestroy:      testAccCheckScalewayDomainRecordDestroy(tt),
 		Steps: []resource.TestStep{
-			{
```
Comment (jerjako) [inline scaleway/resource_domain_record_test.go:311]: we should keep this test for dynamic record: `http_service`

Comment (remyleone) [inline scaleway/resource_domain_record_test.go:311]: Why is this test removed?

File: scaleway/helpers_domain.go [hunk 2]
```diff
@@ -30,8 +30,8 @@ func flattenDomainData(data string, recordType domain.RecordType) interface{} {
 		if len(dataSplit) == 2 {
 			return dataSplit[1]
 		}
-	default:
-		return data
+	case domain.RecordTypeTXT:
```
Comment (remyleone) [inline scaleway/helpers_domain.go:33]: Could we return directly in the switch clause? It seems easier to follow

File: scaleway/resource_domain_record.go [hunk 3]
```diff
@@ -261,11 +263,45 @@ func resourceScalewayDomainRecordCreate(ctx context.Context, d *schema.ResourceD
...
+		DNSZone: dnsZone,
+		Name:    d.Get("name").(string),
+		Type:    recordType,
+	}, scw.WithAllPages(), scw.WithContext(ctx))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var currentRecord *domain.Record
```
Comment (remyleone) [inline scaleway/resource_domain_record.go:286]: Could this be extracted in a function?

File: scaleway/resource_domain_record.go [hunk 4]
```diff
@@ -299,9 +335,8 @@ func resourceScalewayDomainRecordRead(ctx context.Context, d *schema.ResourceDat
 			return diag.FromErr(err)
 		}
 
-		for _, r := range res.Records {
-			record = r
-			break
+		if len(res.Records) > 0 {
```
Comment (remyleone) [inline scaleway/resource_domain_record.go:338]: Why did this part changed?

Comment (Monitob) [reply scaleway/resource_domain_record.go:338]: We were doing un unnecessary loop on Records.

File: scaleway/resource_domain_record_test.go (hunk 1 above)
Comment (Monitob) [reply scaleway/resource_domain_record_test.go]: Test just added

File: scaleway/resource_domain_record.go (hunk 3 above)
Comment (Monitob) [reply scaleway/resource_domain_record.go]: just added on helpers`getRecordFromData`

File: scaleway/helpers_domain.go (hunk 2 above)
Comment (Monitob) [reply scaleway/helpers_domain.go:34]: ok good 👍 idea

File: scaleway/resource_domain_record.go [hunk 5]
```diff
@@ -382,34 +402,64 @@ func resourceScalewayDomainRecordRead(ctx context.Context, d *schema.ResourceDat
 func resourceScalewayDomainRecordUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	domainAPI := newDomainAPI(meta)
 
-	if d.HasChangesExcept("dns_zone", "keep_empty_zone", "name", "type") {
-		id := d.Id()
+	recordID := d.Id()
+	updateRequest := &domain.UpdateDNSZoneRecordsRequest{}
```
Comment (remyleone) [inline scaleway/resource_domain_record.go:406]: Could you write it in a litteral form that is easier to read?

**Review by remyleone — APPROVED** [review]

**Pull request merged**